### DISCUSSION TITLE
How do i get access-token from local storage is SSR without cookie?

### DISCUSSION THREAD

**Discussion by pieeee** [vercel/next.js#39915] [Help] [answered]
The backend is bound to send an access token in my application when a user signs in, which I have to store in local storage/session. Is there any way to get the token from local storage to SSR?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I will open saying no, because the local storage is a client side construct that's not shared as part of any requests headers, naturally anyway.

For a `getServerSideProps` request, you won't see the storage as part of the incoming headers. 

Of course, for any fetch requests you craft yourself you can include local storage values as headers or payload. This effectively means serializing the stored value, by hand.

The root of the issue is that the server is a totally independent entity than the client. And the local storage is never shared.

I'd say, that your back-end should place a cookie on the browser, and use that for SSR user related work.
